compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -702,6 +702,20 @@ declare_lint! {
     /// `PhantomData`.
     ///
     /// Otherwise consider removing the unused code.
+    ///
+    /// ### Limitations
+    ///
+    /// Removing fields that are only used for side-effects and never
+    /// read will result in behavioral changes. Examples of this
+    /// include:
+    ///
+    /// - If a field's value performs an action when it is dropped.
+    /// - If a field's type does not implement an auto trait
+    ///   (e.g. `Send`, `Sync`, `Unpin`).
+    ///
+    /// For side-effects from field values, this lint should be
+    /// allowed on those fields. For side-effects from field types,
```

Comment:
```suggestion
    /// allowed on those fields. For side-effects from <VERB> field types,
```

idk what should go here but it feels like it could use a verb 😆